src/lib.rs

```diff
@@ -1,4 +1,99 @@
 mod client;
 mod util;
+use futures::{future, stream::Stream, Async, Future, Poll};
+use http::{Request, Response};
+use hyper::{
+    client::{HttpConnector, ResponseFuture},
+    Body, Client as HyperClient, body::Payload
+};
+use bytes::Bytes;
+use tower_service::Service;
+use tower_retry::Policy;
 
 pub use self::client::{Connect, Connection};
+
+struct Client<C = HttpConnector, B = Body> {
+    inner: HyperClient<C, B>
+}
+
+impl Default for Client<HttpConnector, Body> {
+    fn default() -> Self {
+        Self {
+            inner: HyperClient::new()
```

Comment:
I would change this to new and actually why not just accept a `Client<C, B: Payload>` and just provide the service wrapper?

Reply:
> I would change this to new and actually why not just accept a Client<C, B: Payload> and just provide the service wrapper?

I'm not entirely sure why I wrote this. I agree that this would be a better change.

Reply:
fixed, but i'm unsure of how to best handle the generic parameters. mimic hyper's client?